YarpJS/src/YarpJS_Image.cpp

```diff
@@ -36,6 +49,17 @@ void YarpJS_Image::compress(int compression_quality)
       encodeString = ".jpg";
     }
 
+    int imgPixelCode = this->getYarpObj()->getPixelCode();
+    if (imgPixelCode != VOCAB_PIXEL_BGR
+        && imgPixelCode != VOCAB_PIXEL_BGRA) {
+        auto conversionCodePtr = yarpCode2colorConvCode.find(imgPixelCode);
+        if (conversionCodePtr != yarpCode2colorConvCode.end()) {
+            cv::cvtColor(internalImage, internalImage, conversionCodePtr->second, 0);
+        } else {
+            throw std::out_of_range("YarpJS_Image::compress(int compression_quality): pixel code not supported");
+        }
```

Comment:
I think that in order to slightly improve the readability, the else branch can be removed:

```suggestion
        if (conversionCodePtr == yarpCode2colorConvCode.end()) {
            throw std::out_of_range("YarpJS_Image::compress(int compression_quality): pixel code not supported");    
        } 
        cv::cvtColor(internalImage, internalImage, conversionCodePtr->second, 0);
```

Also I would avoid throwing `std::out_of_range` as it is commonly related to bad access to container objects. I think [`std::logic_error`](https://en.cppreference.com/w/cpp/error/logic_error) or [`std::domain_error`](https://en.cppreference.com/w/cpp/error/domain_error) could be a better fit for this case.

Reply:
Regarding the removal of `else`, absolutely.
Regarding the exception type, I kind of disagree:
- `std::logic_error` seems too generic to me, all the other types being derived from this one, so I really wouldn't pick this one.
- `std::domain_error` is suitable for an argument being out of range for a given operation (e.g. [math_errhandling](https://en.cppreference.com/w/cpp/numeric/math/math_errhandling)).
- **`std::out_of_range`** is actually the default exception used by `std::map` when you try to access a non existing key and that is why I picked it.

As you mentioned yourself, `std::out_of_range` is suited for bad access to containers. More precisely, it indicates an "attempt to access elements out of defined range." (refer to documentation). Well, at a lower level that's the case here. Then the custom message that follows brings some clarification.

Reply:
As per what we agreed in our live discussion, I'm keeping your suggestion on the `<if/else>`, but keeping `std::out_of_range`.